pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1186,6 +1186,27 @@ self: super: {
     '';
   }) super.cryptol;
 
+  # Need to use the latest upstream for compatbility with newer Z3 versions.
+  z3 = overrideCabal (drv: {
+    version = "411.0.1-pre";
+
+    src = pkgs.fetchFromGitHub {
+      owner = "IagoAbal";
+      repo = "haskell-z3";
+      # https://github.com/IagoAbal/haskell-z3/pull/88
+      rev = "b77a17e5eeb7db82656bcbcd66c6e952207e69ca";
+      hash = "sha256-spYijACiJFkIwLYV6NqQ7IJwNGtI/eaaIuWIIhNb6oY=";
+    };
```

Comment:
Is it possible *just* to apply the patch. The build would then fail when upstream makes a new release and we'd automatically know we can remove this override.

Reply:
I tried it, there are unfortunately too many changes in between and almost all henks failed to apply.

Reply:
Incidentally, this solution is now more complicated than the original overrides to the z3_* boolean types, and we're also adding more dependencies on older z3 versions. I'm beginning to be more of a fan of the NIX_CFLAGS_COMPILE solution especially given that all the test suites pass and we can continue to track hackage.

Reply:
It occurred to me that we shouldn't have passed this to GHC, but instead to `hsc2hs` (I briefly checked z3 and it uses that). In theory `--hsc2hs-option=-D…` should work, then.

Reply:
Ahhh! I'll try that, good idea.